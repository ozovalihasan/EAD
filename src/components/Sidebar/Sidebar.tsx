import useCustomizationStore from '@/zustandStore/customizationStore';
import useStore from '@/zustandStore/store';

export const Sidebar = () => {

  const { onChangeAssociationType, 
    onTableNameChange, 
    addTable, 
    addAttribute, 
    removeAttribute,
    removeTable,
    onAttributeNameChange,
    onAttributeTypeChange,
  } = useStore();

  const {
    locationSidebar,
    widthSidebar,
    toggleLocationSidebar,
    toggleTextMode,
  } = useCustomizationStore()
  

  const onDragStart = (
    event: React.DragEvent<HTMLDivElement> , 
    nodeType: string,
    tableId: string,
  ) => {
    event.dataTransfer.setData('application/reactflow', nodeType);
    event.dataTransfer.setData('tableId', tableId);
    event.dataTransfer.effectAllowed = 'move';
    event.dataTransfer.setDragImage(event.target as Element, 0, 0);
  };

  let tables = useStore((state) => state.tables);
  const showTextOnEdges = useCustomizationStore(store => store.showTextOnEdges)
  const sidebarVisible = useCustomizationStore(store => store.sidebarVisible)

  return (
    <div>
      <aside className={`h-full relative py-3 px-2 overflow-y-scroll ${sidebarVisible || 'hidden'}`} style={{width: (locationSidebar === "left" ? widthSidebar : (window.innerWidth - (widthSidebar as number))) as number}} >
        
        {
          
          Object.keys(tables).map((tableId: string) => {
            return (
              <div className="my-4 w-full bg-transparent border border-first-500 border-solid rounded-md" key={tableId} onDragStart={(event) => onDragStart(event, 'default', tableId)} draggable>
                <input className="p-2 rounded-md w-full" placeholder='Table name' type="text" value={tables[tableId].name} onChange={(event) => onTableNameChange(event, tableId)} />
                {
                  Object.keys(tables[tableId].attributes).map((attributeId) => {
                    return (
                      <div className="flex m-1" key={attributeId} >
                        <button className="btn-first rounded-full aspect-square h-6" title="Remove the attribute" onClick={event => removeAttribute(event, tableId, attributeId)} >-</button>
                        <input className="p-1 w-2/3 rounded-md" placeholder="Attribute" type="text" value={tables[tableId].attributes[attributeId].name} onChange={(event) => onAttributeNameChange(event, tableId, attributeId)} />
                        <select
                          className="w-1/3"
                          value={tables[tableId].attributes[attributeId].type}
                          onChange={(event) => onAttributeTypeChange(event, tableId, attributeId)}
                        >
                          {['primary_key', 'string', 'text', 'integer', 'float', 'decimal', 'datetime', 'timestamp',
                            'time', 'date', 'binary', 'boolean', 'references'].map((item) => (
                              <option
                                key={item}
                                value={item}
                              >
                                {item}
                              </option>
                          ))}

                        </select>
                      </div>
                    )
                  })
                }
                <button className="btn-first rounded-full m-1 aspect-square h-6" title="Add an attribute" onClick={event => addAttribute(event, tableId)}>+</button>
                <button className="right-0 absolute -translate-x-1/2 translate-y-1/2 btn-first rounded-full aspect-square h-6" title="Delete the table" onClick={event => removeTable(event, tableId)}>-</button>
              </div>
            )
          })
        }
        <button className="p-1 btn-first rounded-full aspect-square h-10" title="Add a table" onClick={addTable}>+</button>

        
        <details open className='mt-8 [&>summary>span:nth-child(1)]:open:hidden [&>summary>span:nth-child(2)]:open:inline '>
          <summary className='btn-third'>
            <span>Show </span>
            <span className='hidden'>Hide </span> 
            menu
          </summary>
          <div className='mt-8 flex-col flex'>
            <label className='flex mb-5 cursor-pointer'>
              <input className='mr-6' type="checkbox" onChange={toggleTextMode}/>
              Show Association Names
            </label>
            <label className='flex mb-5 cursor-pointer'>
              <input className='mr-6 ' type="checkbox" checked={ locationSidebar === "right"} onChange={toggleLocationSidebar}/>
              Show the sidebar at the right of the window
            </label>
            

                
          </div>
        </details>
      </aside>
    </div>
  );
};
