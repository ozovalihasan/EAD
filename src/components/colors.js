const colors = {
  suitable: '#F4FECD',
  disabled: '#CFDBD5',
  association: '#FFF',
  attributeContainer: '#AAFAE9',
  attribute: '#AAFAE9',
  entity: '#94F6EA',
  entityContainer: '#94F6EA',
  EAD: '#FFF',
  factory: '#FFF',
  entityClone: '#F5C987',
  chosen: '#52C9CF',
  warning: '#F94144',
};

export default colors;
