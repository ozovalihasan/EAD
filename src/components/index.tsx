export * from './Markers/CrossMarker'
export * from './Markers/CrowsFootMarker'
export * from './Markers/CircleLineMarker'
export * from './Markers/StraightLineMarker'
export * from './TargetHandle/TargetHandle'
export * from './SelectThroughNode/SelectThroughNode'
export * from './CustomHandle/CustomHandle'
export * from './ConnectionLine/ConnectionLine'
export * from './HasAnyEdge/HasAnyEdge';
export * from './HasManyEdge/HasManyEdge';
export * from './HasOneEdge/HasOneEdge';
export * from './ThroughEdge/ThroughEdge';
export * from './Sidebar/Sidebar';
export * from './Markers/MarkerBase';
export * from './EntityNode/EntityNode';
export * from './HasManyHandle/HasManyHandle';
export * from './HasOneHandle/HasOneHandle';
export * from './TableName/TableName';
export * from './ThroughHandle/ThroughHandle';
export * from './RemoveEdgeButton/RemoveEdgeButton';
export * from './RemoveNodeEdgeButton/RemoveNodeEdgeButton';
export * from './ToggleOptionalButton/ToggleOptionalButton';
export * from './ShowEdgeText/ShowEdgeText';
export * from './Navbar/Navbar';
export * from './EADLogo/EADLogo';
export * from './GithubLogo/GithubLogo';
export * from './AllHandlers/AllHandlers';
export * from './Settings/Settings';
export * from './App/App';
export * from './SidebarHandle/SidebarHandle';
export * from './MainReactFlow/MainReactFlow';
export * from './AttributeTypeOptions/AttributeTypeOptions';
export * from './SidebarOptions/SidebarOptions';
export * from './MainColorDropdown/MainColorDropdown';
export * from './TableOptions/TableOptions';
export * from './TableAttributes/TableAttributes';
export * from './AddTableButton/AddTableButton';
export * from './AddAttributeButton/AddAttributeButton';
export * from './RemoveAttributeButton/RemoveAttributeButton';
export * from './ReorderAllTablesButton/ReorderAllTablesButton';
export * from './ProjectName/ProjectName';
