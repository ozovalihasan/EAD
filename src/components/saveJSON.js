/* eslint-disable no-console */
const saveJSON = (data, filename) => {
  let dataNew = data;
  let filenameNew = filename;
  if (!dataNew) {
    console.error('No data');
    return;
  }

  if (!filenameNew) { filenameNew = 'console.json'; }

  if (typeof dataNew === 'object') {
    dataNew = JSON.stringify(dataNew, undefined, 4);
  }

  const blob = new Blob([dataNew], { type: 'text/json' });
  const e = document.createEvent('MouseEvents');
  const a = document.createElement('a');

  a.download = filenameNew;
  a.href = window.URL.createObjectURL(blob);
  a.dataset.downloadurl = ['text/json', a.download, a.href].join(':');
  e.initMouseEvent('click', true, false, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
  a.dispatchEvent(e);
};

export default saveJSON;
