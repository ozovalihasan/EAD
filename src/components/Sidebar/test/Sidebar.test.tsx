import {  Sidebar } from '../Sidebar';
import { render, screen, renderHook, fireEvent } from "@testing-library/react";
import useStore from '@/zustandStore/store';
import useCustomizationStore from '@/zustandStore/customizationStore';
import { SidebarOptionsType } from '@/components';

jest.mock("@/components", () => ({
  Settings: () => (
    <>
      MockSettings
    </>
  ),
  AttributeTypeOptions: () => (
    <>
      MockAttributeTypeOptions
    </>
  ),
  SidebarOptions: (props: SidebarOptionsType) => (
    <>
      MockSidebarOptions
      { 
        Object.keys(props)
          .map((key) => `${key}: ${props[key as keyof typeof props]}`) 
      }
    </>
  ),
}))

jest.mock("@/icons", () => ({
  PlusSign: () => (
    <>
      MockPlusSign
    </>
  ),
  MinusSign: () => (
    <>
      MockMinusSign
    </>
  ),
}))


let renderReadyComponent: JSX.Element;

beforeEach(() => {
    
  useStore.setState({ 
    tables: {
      "1": {
        "name": "MockFirstTable",
        "attributes": {
          "13": {
            "name": "mockFirstAttribute",
            "type": "string"
          },
          "14": {
            "name": "mockSecondAttribute",
            "type": "string"
          }
        }, 
        "superclassId": ""
      },
      "2": {
        "name": "MockSecondTable",
        "attributes": {
          "15": {
            "name": "mockThirdAttribute",
            "type": "string"
          },
          "16": {
            "name": "mockFourthAttribute",
            "type": "string"
          }
        },
        "superclassId": ""
      },
    },
    resetStore: jest.fn(),
    uploadStore: jest.fn(),
    addAttribute: jest.fn(),
    removeAttribute: jest.fn(),
    removeTable: jest.fn(),
    addTable: jest.fn(),
    changeTableSuperClass: jest.fn(),
    onAttributeNameChange: jest.fn(),
  })
    
  renderReadyComponent = (
    <Sidebar />
  );
});

describe('<Sidebar />', () => {
  
  describe('if the sidebar is visible', () => {

    beforeEach(() => {
      useCustomizationStore.setState({ 
        sidebarVisible: true
      })
    });

    it('renders all tables', () => {

      render(renderReadyComponent );

      expect(screen.getByDisplayValue(/MockFirstTable/i)).toBeInTheDocument();
      expect(screen.getByDisplayValue(/MockSecondTable/i)).toBeInTheDocument();
    });

    it('renders all attributes of all tables', () => {

      render(renderReadyComponent );

      expect(screen.getByDisplayValue(/mockFirstAttribute/i)).toBeInTheDocument();
      expect(screen.getByDisplayValue(/mockSecondAttribute/i)).toBeInTheDocument();
      expect(screen.getByDisplayValue(/mockThirdAttribute/i)).toBeInTheDocument();
      expect(screen.getByDisplayValue(/mockFourthAttribute/i)).toBeInTheDocument();
    });

    it('renders the necessary buttons', () => {

      render(renderReadyComponent );
      const { result } = renderHook(() => useStore());


      const addAttributeButtons = screen.getAllByTitle(/Add an attribute/i)
      expect(addAttributeButtons.length).toBe(2);
      fireEvent.click(addAttributeButtons[0])
      expect(result.current.addAttribute).toHaveBeenCalledTimes(1);
      
      const removeAttributeButtons = screen.getAllByTitle(/Remove the attribute/i)
      expect(removeAttributeButtons.length).toBe(4);
      fireEvent.click(removeAttributeButtons[0])
      expect(result.current.removeAttribute).toHaveBeenCalledTimes(1);
      
      const deleteTableButtons = screen.getAllByTitle(/Delete the table/i)
      expect(deleteTableButtons.length).toBe(2);
      fireEvent.click(deleteTableButtons[0])
      expect(result.current.removeAttribute).toHaveBeenCalledTimes(1);

      const addTableButton = screen.getByTitle(/Add a table/i)
      expect(addTableButton).toBeInTheDocument();
      fireEvent.click(addTableButton)
      expect(result.current.addTable).toHaveBeenCalledTimes(1);
    });

    it('renders another component', () => {

      render(renderReadyComponent );
      
      expect(screen.getByText(/MockSettings/i)).toBeInTheDocument();
      expect(screen.getAllByText(/AttributeTypeOptions/i).length).toBe(4);
    });

    it('renders a dropdown to select a superclass for each table', () => {

      render(renderReadyComponent);
      
      expect(screen.getAllByText(/MockSidebarOptions/i).length).toBe(2);
    });
    
    it('renders an input element for each attribute', () => {

      render(renderReadyComponent );
      const { result } = renderHook(() => useStore());

      const inputElement = screen.getByDisplayValue(/mockFirstAttribute/i)
      
      fireEvent.change(inputElement, {target: {value: 'mockAttributeName'}})

      expect(result.current.onAttributeNameChange).toHaveBeenCalledTimes(1);
    });
    
    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
  })  

  describe('if the sidebar is invisible', () => {


    beforeEach(() => {
      useCustomizationStore.setState({ 
        sidebarVisible: false
      })
    });

    it('has a "hidden" class', () => {
      render(renderReadyComponent );

      const asideElement = screen.getAllByText("")[0].getElementsByTagName("aside")[0]
      
      expect(asideElement.classList).toContain("hidden");
    });


    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
  })

});