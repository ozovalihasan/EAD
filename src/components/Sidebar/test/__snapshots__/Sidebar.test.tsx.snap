// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> if the sidebar is invisible renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <aside
        class="h-full relative py-3 px-2 overflow-y-scroll hidden"
        style="width: 204px;"
      >
        <div
          class="my-4 w-full bg-transparent border border-first-500 border-solid rounded-md"
          draggable="true"
        >
          <input
            class="p-2 rounded-md w-full"
            placeholder="Table name"
            type="text"
            value="MockFirstTable"
          />
          <div
            class="flex m-1"
          >
            <button
              class="btn-first rounded-full aspect-square h-6"
              title="Remove the attribute"
            >
              <div
                class="stroke-[40] w-3 h-3"
              >
                MockMinusSign
              </div>
            </button>
            <input
              class="p-1 w-2/3 rounded-md"
              placeholder="Attribute"
              type="text"
              value="mockFirstAttribute"
            />
            <div
              class="w-1/3"
            >
              MockAttributeTypeOptions
            </div>
          </div>
          <div
            class="flex m-1"
          >
            <button
              class="btn-first rounded-full aspect-square h-6"
              title="Remove the attribute"
            >
              <div
                class="stroke-[40] w-3 h-3"
              >
                MockMinusSign
              </div>
            </button>
            <input
              class="p-1 w-2/3 rounded-md"
              placeholder="Attribute"
              type="text"
              value="mockSecondAttribute"
            />
            <div
              class="w-1/3"
            >
              MockAttributeTypeOptions
            </div>
          </div>
          <button
            class="btn-first rounded-full m-1 aspect-square h-6"
            title="Add an attribute"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockPlusSign
            </div>
          </button>
          <button
            class="right-0 absolute -translate-x-full -translate-y-1/2 btn-first rounded-full aspect-square h-6 "
            title="Delete the table"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockMinusSign
            </div>
          </button>
        </div>
        <div
          class="my-4 w-full bg-transparent border border-first-500 border-solid rounded-md"
          draggable="true"
        >
          <input
            class="p-2 rounded-md w-full"
            placeholder="Table name"
            type="text"
            value="MockSecondTable"
          />
          <div
            class="flex m-1"
          >
            <button
              class="btn-first rounded-full aspect-square h-6"
              title="Remove the attribute"
            >
              <div
                class="stroke-[40] w-3 h-3"
              >
                MockMinusSign
              </div>
            </button>
            <input
              class="p-1 w-2/3 rounded-md"
              placeholder="Attribute"
              type="text"
              value="mockThirdAttribute"
            />
            <div
              class="w-1/3"
            >
              MockAttributeTypeOptions
            </div>
          </div>
          <div
            class="flex m-1"
          >
            <button
              class="btn-first rounded-full aspect-square h-6"
              title="Remove the attribute"
            >
              <div
                class="stroke-[40] w-3 h-3"
              >
                MockMinusSign
              </div>
            </button>
            <input
              class="p-1 w-2/3 rounded-md"
              placeholder="Attribute"
              type="text"
              value="mockFourthAttribute"
            />
            <div
              class="w-1/3"
            >
              MockAttributeTypeOptions
            </div>
          </div>
          <button
            class="btn-first rounded-full m-1 aspect-square h-6"
            title="Add an attribute"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockPlusSign
            </div>
          </button>
          <button
            class="right-0 absolute -translate-x-full -translate-y-1/2 btn-first rounded-full aspect-square h-6 "
            title="Delete the table"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockMinusSign
            </div>
          </button>
        </div>
        <button
          class="p-1 btn-first rounded-full aspect-square h-10"
          title="Add a table"
        >
          <div
            class="stroke-[40] w-5 h-5"
          >
            MockPlusSign
          </div>
        </button>
        MockSettings
      </aside>
    </div>
  </body>,
  "container": <div>
    <aside
      class="h-full relative py-3 px-2 overflow-y-scroll hidden"
      style="width: 204px;"
    >
      <div
        class="my-4 w-full bg-transparent border border-first-500 border-solid rounded-md"
        draggable="true"
      >
        <input
          class="p-2 rounded-md w-full"
          placeholder="Table name"
          type="text"
          value="MockFirstTable"
        />
        <div
          class="flex m-1"
        >
          <button
            class="btn-first rounded-full aspect-square h-6"
            title="Remove the attribute"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockMinusSign
            </div>
          </button>
          <input
            class="p-1 w-2/3 rounded-md"
            placeholder="Attribute"
            type="text"
            value="mockFirstAttribute"
          />
          <div
            class="w-1/3"
          >
            MockAttributeTypeOptions
          </div>
        </div>
        <div
          class="flex m-1"
        >
          <button
            class="btn-first rounded-full aspect-square h-6"
            title="Remove the attribute"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockMinusSign
            </div>
          </button>
          <input
            class="p-1 w-2/3 rounded-md"
            placeholder="Attribute"
            type="text"
            value="mockSecondAttribute"
          />
          <div
            class="w-1/3"
          >
            MockAttributeTypeOptions
          </div>
        </div>
        <button
          class="btn-first rounded-full m-1 aspect-square h-6"
          title="Add an attribute"
        >
          <div
            class="stroke-[40] w-3 h-3"
          >
            MockPlusSign
          </div>
        </button>
        <button
          class="right-0 absolute -translate-x-full -translate-y-1/2 btn-first rounded-full aspect-square h-6 "
          title="Delete the table"
        >
          <div
            class="stroke-[40] w-3 h-3"
          >
            MockMinusSign
          </div>
        </button>
      </div>
      <div
        class="my-4 w-full bg-transparent border border-first-500 border-solid rounded-md"
        draggable="true"
      >
        <input
          class="p-2 rounded-md w-full"
          placeholder="Table name"
          type="text"
          value="MockSecondTable"
        />
        <div
          class="flex m-1"
        >
          <button
            class="btn-first rounded-full aspect-square h-6"
            title="Remove the attribute"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockMinusSign
            </div>
          </button>
          <input
            class="p-1 w-2/3 rounded-md"
            placeholder="Attribute"
            type="text"
            value="mockThirdAttribute"
          />
          <div
            class="w-1/3"
          >
            MockAttributeTypeOptions
          </div>
        </div>
        <div
          class="flex m-1"
        >
          <button
            class="btn-first rounded-full aspect-square h-6"
            title="Remove the attribute"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockMinusSign
            </div>
          </button>
          <input
            class="p-1 w-2/3 rounded-md"
            placeholder="Attribute"
            type="text"
            value="mockFourthAttribute"
          />
          <div
            class="w-1/3"
          >
            MockAttributeTypeOptions
          </div>
        </div>
        <button
          class="btn-first rounded-full m-1 aspect-square h-6"
          title="Add an attribute"
        >
          <div
            class="stroke-[40] w-3 h-3"
          >
            MockPlusSign
          </div>
        </button>
        <button
          class="right-0 absolute -translate-x-full -translate-y-1/2 btn-first rounded-full aspect-square h-6 "
          title="Delete the table"
        >
          <div
            class="stroke-[40] w-3 h-3"
          >
            MockMinusSign
          </div>
        </button>
      </div>
      <button
        class="p-1 btn-first rounded-full aspect-square h-10"
        title="Add a table"
      >
        <div
          class="stroke-[40] w-5 h-5"
        >
          MockPlusSign
        </div>
      </button>
      MockSettings
    </aside>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Sidebar /> if the sidebar is visible renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <aside
        class="h-full relative py-3 px-2 overflow-y-scroll true"
        style="width: 204px;"
      >
        <div
          class="my-4 w-full bg-transparent border border-first-500 border-solid rounded-md"
          draggable="true"
        >
          <input
            class="p-2 rounded-md w-full"
            placeholder="Table name"
            type="text"
            value="MockFirstTable"
          />
          <div
            class="flex m-1"
          >
            <button
              class="btn-first rounded-full aspect-square h-6"
              title="Remove the attribute"
            >
              <div
                class="stroke-[40] w-3 h-3"
              >
                MockMinusSign
              </div>
            </button>
            <input
              class="p-1 w-2/3 rounded-md"
              placeholder="Attribute"
              type="text"
              value="mockFirstAttribute"
            />
            <div
              class="w-1/3"
            >
              MockAttributeTypeOptions
            </div>
          </div>
          <div
            class="flex m-1"
          >
            <button
              class="btn-first rounded-full aspect-square h-6"
              title="Remove the attribute"
            >
              <div
                class="stroke-[40] w-3 h-3"
              >
                MockMinusSign
              </div>
            </button>
            <input
              class="p-1 w-2/3 rounded-md"
              placeholder="Attribute"
              type="text"
              value="mockSecondAttribute"
            />
            <div
              class="w-1/3"
            >
              MockAttributeTypeOptions
            </div>
          </div>
          <button
            class="btn-first rounded-full m-1 aspect-square h-6"
            title="Add an attribute"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockPlusSign
            </div>
          </button>
          <button
            class="right-0 absolute -translate-x-full -translate-y-1/2 btn-first rounded-full aspect-square h-6 "
            title="Delete the table"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockMinusSign
            </div>
          </button>
        </div>
        <div
          class="my-4 w-full bg-transparent border border-first-500 border-solid rounded-md"
          draggable="true"
        >
          <input
            class="p-2 rounded-md w-full"
            placeholder="Table name"
            type="text"
            value="MockSecondTable"
          />
          <div
            class="flex m-1"
          >
            <button
              class="btn-first rounded-full aspect-square h-6"
              title="Remove the attribute"
            >
              <div
                class="stroke-[40] w-3 h-3"
              >
                MockMinusSign
              </div>
            </button>
            <input
              class="p-1 w-2/3 rounded-md"
              placeholder="Attribute"
              type="text"
              value="mockThirdAttribute"
            />
            <div
              class="w-1/3"
            >
              MockAttributeTypeOptions
            </div>
          </div>
          <div
            class="flex m-1"
          >
            <button
              class="btn-first rounded-full aspect-square h-6"
              title="Remove the attribute"
            >
              <div
                class="stroke-[40] w-3 h-3"
              >
                MockMinusSign
              </div>
            </button>
            <input
              class="p-1 w-2/3 rounded-md"
              placeholder="Attribute"
              type="text"
              value="mockFourthAttribute"
            />
            <div
              class="w-1/3"
            >
              MockAttributeTypeOptions
            </div>
          </div>
          <button
            class="btn-first rounded-full m-1 aspect-square h-6"
            title="Add an attribute"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockPlusSign
            </div>
          </button>
          <button
            class="right-0 absolute -translate-x-full -translate-y-1/2 btn-first rounded-full aspect-square h-6 "
            title="Delete the table"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockMinusSign
            </div>
          </button>
        </div>
        <button
          class="p-1 btn-first rounded-full aspect-square h-10"
          title="Add a table"
        >
          <div
            class="stroke-[40] w-5 h-5"
          >
            MockPlusSign
          </div>
        </button>
        MockSettings
      </aside>
    </div>
  </body>,
  "container": <div>
    <aside
      class="h-full relative py-3 px-2 overflow-y-scroll true"
      style="width: 204px;"
    >
      <div
        class="my-4 w-full bg-transparent border border-first-500 border-solid rounded-md"
        draggable="true"
      >
        <input
          class="p-2 rounded-md w-full"
          placeholder="Table name"
          type="text"
          value="MockFirstTable"
        />
        <div
          class="flex m-1"
        >
          <button
            class="btn-first rounded-full aspect-square h-6"
            title="Remove the attribute"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockMinusSign
            </div>
          </button>
          <input
            class="p-1 w-2/3 rounded-md"
            placeholder="Attribute"
            type="text"
            value="mockFirstAttribute"
          />
          <div
            class="w-1/3"
          >
            MockAttributeTypeOptions
          </div>
        </div>
        <div
          class="flex m-1"
        >
          <button
            class="btn-first rounded-full aspect-square h-6"
            title="Remove the attribute"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockMinusSign
            </div>
          </button>
          <input
            class="p-1 w-2/3 rounded-md"
            placeholder="Attribute"
            type="text"
            value="mockSecondAttribute"
          />
          <div
            class="w-1/3"
          >
            MockAttributeTypeOptions
          </div>
        </div>
        <button
          class="btn-first rounded-full m-1 aspect-square h-6"
          title="Add an attribute"
        >
          <div
            class="stroke-[40] w-3 h-3"
          >
            MockPlusSign
          </div>
        </button>
        <button
          class="right-0 absolute -translate-x-full -translate-y-1/2 btn-first rounded-full aspect-square h-6 "
          title="Delete the table"
        >
          <div
            class="stroke-[40] w-3 h-3"
          >
            MockMinusSign
          </div>
        </button>
      </div>
      <div
        class="my-4 w-full bg-transparent border border-first-500 border-solid rounded-md"
        draggable="true"
      >
        <input
          class="p-2 rounded-md w-full"
          placeholder="Table name"
          type="text"
          value="MockSecondTable"
        />
        <div
          class="flex m-1"
        >
          <button
            class="btn-first rounded-full aspect-square h-6"
            title="Remove the attribute"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockMinusSign
            </div>
          </button>
          <input
            class="p-1 w-2/3 rounded-md"
            placeholder="Attribute"
            type="text"
            value="mockThirdAttribute"
          />
          <div
            class="w-1/3"
          >
            MockAttributeTypeOptions
          </div>
        </div>
        <div
          class="flex m-1"
        >
          <button
            class="btn-first rounded-full aspect-square h-6"
            title="Remove the attribute"
          >
            <div
              class="stroke-[40] w-3 h-3"
            >
              MockMinusSign
            </div>
          </button>
          <input
            class="p-1 w-2/3 rounded-md"
            placeholder="Attribute"
            type="text"
            value="mockFourthAttribute"
          />
          <div
            class="w-1/3"
          >
            MockAttributeTypeOptions
          </div>
        </div>
        <button
          class="btn-first rounded-full m-1 aspect-square h-6"
          title="Add an attribute"
        >
          <div
            class="stroke-[40] w-3 h-3"
          >
            MockPlusSign
          </div>
        </button>
        <button
          class="right-0 absolute -translate-x-full -translate-y-1/2 btn-first rounded-full aspect-square h-6 "
          title="Delete the table"
        >
          <div
            class="stroke-[40] w-3 h-3"
          >
            MockMinusSign
          </div>
        </button>
      </div>
      <button
        class="p-1 btn-first rounded-full aspect-square h-10"
        title="Add a table"
      >
        <div
          class="stroke-[40] w-5 h-5"
        >
          MockPlusSign
        </div>
      </button>
      MockSettings
    </aside>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
